### PR TITLE
Improved panel resizing

### PR DESCRIPTION
**Pull request by kirlat**
This build introduces improved CSS-only panel resizing. Please see #332 and some comments within this issue for more details.

Please note that I've also changed a layout of mobile version of word usage examples panel. Now it looks like below:
image
It has a single button that is responsible for turning all source links on and off. This is to solve the problem with inability to save individual example fragment's state as we moved them out of Vue SFC. Please see #332.

Would welcome all comments and suggestions about this implementations. I thought it was not possible, but looks like we can do it with pure CSS.

Thanks!

### CODE REVIEW

File: src/vue/components/panel-compact.vue [hunk 1]
```diff
@@ -578,6 +548,9 @@ export default {
         })
         .on('resizemove', event => {
           let target = event.target
+          // Indicate that panel received a custom size
+          console.info(`resizemove`)
```
[the remark this answers is not in the document]
Comment (kirlat) [reply src/vue/components/panel-compact.vue]: Fixed that!

**Comment by irina060981** [comment]
There are some strange behaviour with resizing: 

1) (I was increasing width)
image

Header and footer are increasing and content is not.
Resizing icon is on the same width as header, not content.

2) There is some problem with transparancy on mobile
image
image

**Comment by irina060981** [comment]
3) (increasing width to the right)
image

**Comment by irina060981** [comment]
4) It does not correctly ajust the width for inflections table
image

File: src/vue/components/word-usage-examples/word-usage-examples.vue [hunk 2]
```diff
@@ -11,16 +11,56 @@
...
     <div class="alpheios_word_usage_list_mainblock" v-if="showWordUsageExampleItems">
-      <div v-if="wordUsageListSorted.length > 0">
-        <word-usage-examples-item
-            v-for="wordUsageItem in wordUsageListSorted"
-            v-bind:key="wordUsageItem.ID"
-            :wordUsageItem="wordUsageItem"
-        ></word-usage-examples-item>
-      </div>
-      <div v-else>
+      <template v-if="wordUsageListSorted.length > 0">
```
Comment (irina060981) [inline src/vue/components/word-usage-examples/word-usage-examples.vue]: @kirlat, why did you decide to remove a separate component for the word-usage-item and place all the code inside directly word-usage-examples.vue? I have made such a structure for better independency between parts, for having smaller parts to be easier to maintain and develop.  At the end - to have less code inside one component.
What advantages do you see in decreasing an amount of components and increasing code in one component?

**Comment by kirlat** [comment]
Thanks for reporting the issue, will try to reproduce and fix.

File: src/vue/components/word-usage-examples/word-usage-examples.vue (hunk 2 above)
Comment (kirlat) [reply src/vue/components/word-usage-examples/word-usage-examples.vue]: A separate component as you defined it is definitely better. Unfortunately, I had to remove it due to technical limitation of a component that must have only one root element.

In order to be used in CSS grid, component must produce 5 elements (two link cells, pre and post cells, and a target word cell, see https://github.com/alpheios-project/components/issues/332#issuecomment-486750588). The simplest solution to overcome this limitation I've found is to use `v-for` without a component that can have multiple inner elements.

Please let me know if you need more info on this or if you see a better solution. Thanks!

**Comment by kirlat** [comment]
Hi @irina060981, I've pushed a fix for issues (1),(3), and (4). Please check if it's working for you when you will have time. 

Unfortunately, I cannot reproduce an issue (2). Did you test it in device emulator or on a real device? If that was on a device, what's the URL I can use to test it? 

Thanks!

File: src/vue/components/word-usage-examples/word-usage-examples.vue (hunk 2 above)
Comment (irina060981) [reply src/vue/components/word-usage-examples/word-usage-examples.vue]: Oh, it is an interesting reason - I don't have enough experience with css grid.
I have read all your really rich comments in the issue - thank you for them!

I agree that it is always a best way to use  css-scalling for html layout.
And of course if a separate child component makes it not be possible, than you should move it inside the parent component.

**Comment by irina060981** [comment]
@kirlat , I test resizing with the help of webextension (qa.3.0.0.29) and your branch, because there are some new code in component#master that is absent in your code, so it couldn't be tested with webextension#master or embed-lib#master.
That's why I couldn't test it with embed-lib and publish to the external link.

So I am testing with mobile emulator inside chrome.
For the (2) - I tried to screenshot transparent menu with more details:
image

**Comment by irina060981** [comment]
By the way - some buttons disappered from the toolbar (I tried to test inflection-table resizing and noticed, that the panel has inflections tab, but toolbar doesn't have inflections button.

**Comment by irina060981** [comment]
I could see that resizing is working great (1), (3), (4)!
Only some notices for the inflection-tables:

a) try to select a verb - it shows additional select
image

now it is placed on the next  row, not the same as before

b)try to expand the table and it increases width of the table without increasing width of the window (I believe that it is expected and correct behaviour), but than collapse button appears outside of the user view and it is not really easy to guess how to collapse the table back

image

**Comment by irina060981** [comment]
One small bug:
If you place the panel to the right and try to reduce the width - it will hide close button
image

**Comment by irina060981** [comment]
oh I was able again to reproduce some problems with resizing - here it is a video
https://www.dropbox.com/s/qzsu7ilpztpourd/testPanel2019-04-26.wmv?dl=0

**Comment by kirlat** [comment]
> By the way - some buttons disappered from the toolbar (I tried to test inflection-table resizing and noticed, that the panel has inflections tab, but toolbar doesn't have inflections button.

This was made for #457 to reduce an overall toolbar height. Those buttons are available in the panel, so there is no reason to duplicate them in a toolbar.

**Comment by kirlat** [comment]
> ) try to select a verb - it shows additional select
> image
> 
> now it is placed on the next row, not the same as before

I think it is showing the same behavior as in the previous build; it was not, I believe, affected by the resizing update. If it does not look right, we should create a separate issue for that as it is probably out of scope for the current PR. @balmas, do we have any guidelines about how this part should look?

**Comment by kirlat** [comment]
> b)try to expand the table and it increases width of the table without increasing width of the window (I believe that it is expected and correct behaviour), but than collapse button appears outside of the user view and it is not really easy to guess how to collapse the table back
> 
> image

The panel does not expand for more than 80% of the viewport. Thus the limited size of the panel is an expected behavior. 

I agree that it will be good to have a collapse/expand button always visible, but if we move it so that it will fit into a visible part of the window then, during scroll, it will be stuck somewhere near a middle of the table which, on my opinion, will look strange when the table is scrolled. 
image
I'm not sure how it should look ideally, but if we want to change that existing behavior, we should probably create a separate issue for that. If it is an issue, it was not introduced by the resizing change, I think. @balmas, what do you think, do we need to change anything here and if we do, how should we do that?

**Comment by kirlat** [comment]
> If you place the panel to the right and try to reduce the width - it will hide close button

This is a good catch! I've updated the minimal width of the panel so that all the buttons will fit into it.

**Comment by kirlat** [comment]
> oh I was able again to reproduce some problems with resizing - here it is several videos
> https://www.dropbox.com/s/qzsu7ilpztpourd/testPanel2019-04-26.wmv?dl=0
> 
> https://www.dropbox.com/s/kn0a8zzwpxzist4/testPanel2019-04-26-2.wmv?dl=0
> 
> Hope it would be helpful!

Thanks for the video, it's very helpful for understanding behavior. Some of it is expected: you cannot manually resize the panel to be wider than 80% of a viewport (@balmas, is that correct?).

I see some issues with scrolling when panel is attached to the right and table may appear to be with scroll position in the middle after a series of actions. I'm not sure how hard it will be to fix and will investigate. @balmas, how critical do you think it is?

**Comment by balmas** [comment]
> > oh I was able again to reproduce some problems with resizing - here it is several videos
> > https://www.dropbox.com/s/qzsu7ilpztpourd/testPanel2019-04-26.wmv?dl=0
> > https://www.dropbox.com/s/kn0a8zzwpxzist4/testPanel2019-04-26-2.wmv?dl=0
> > Hope it would be helpful!
> 
> Thanks for the video, it's very helpful for understanding behavior. Some of it is expected: you cannot manually resize the panel to be wider than 80% of a viewport (@balmas, is that correct?).
> 
> I see some issues with scrolling when panel is attached to the right and table may appear to be with scroll position in the middle after a series of actions. I'm not sure how hard it will be to fix and will investigate. @balmas, how critical do you think it is?

Well, maybe my idea to restrict the size of the panel to 80% of the screen size wasn't a good idea?  We could make that bigger but It think we still will have some possible issues here. Would like to get feedback from @monzug the inflection table expert here.  Also, still don't know what we are going to do about all of this on mobile ...

**Comment by kirlat** [comment]
>  Also, still don't know what we are going to do about all of this on mobile ...

On mobile, it is not resizable as I think it will be impossible to drag a thin border with a finger that will cover the line (it's not even super easy to do with the mouse now if the scroll bar interferes). I believe that's why resizing is not used on mobile often. If we want to make it resizable on mobile, we should probably provide some handle large enough to be used with fingers, but that will occupy valuable screen estate. Not sure if we need to do that (but I was thinking about a handle for desktop sometimes because with a scrollbar the latter intercepts mouse action so maybe a handle at the outer side of it would makes sense).

Right now panel on mobile takes half of a screen. I was thinking that for working with dense word data (inflection tables can be a good example) in might make sense to allow users to expand it to occupy the whole screen. So we might have some control that will switch it between 50% and 100% width (something similar to "normal/maximized" window button in most operating systems). That will be a more compact solution than a drag handle, and probably more meaningful.

What do you think?

**Comment by balmas** [comment]
> Right now panel on mobile takes half of a screen. I was thinking that for working with dense word data (inflection tables can be a good example) in might make sense to allow users to expand it to occupy the whole screen. So we might have some control that will switch it between 50% and 100% width (something similar to "normal/maximized" window button in most operating systems). That will be a more compact solution than a drag handle, and probably more meaningful.
> 
> What do you think?

Yes, I definitely want that. I think we have an issue for it somewhere, but if not, we should. So you're right it's not resizing that is an issue per-se on mobile, but designs for scrolling when the data is too big for the displayable window will be.

**Comment by kirlat** [comment]
I cannot reproduce (2), but I think it's probably does not matter because now I'm working on #339 that, if successful, can replace the mobile menu. If not, we can move it into a separate issue because it is not related to resizing.

**Pull request merged**